### PR TITLE
Servicenow assigned user fix

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/46099

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

File: Packs/ServiceNow/ReleaseNotes/2_2_23.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### ServiceNow v2
+- Fixed an issue where mirroring stopped working when invalid user was assigned to the ticket.
```
Comment (bakatzir) [inline Packs/ServiceNow/ReleaseNotes/2_2_23.md:4]:
```suggestion
- Fixed an issue where mirroring stopped working when an invalid user was assigned to the ticket.
```

**Comment by Itay4** [comment]
please add unit test

**Comment by Shellyber** [comment]
@Itay4 Please review

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 2]
```diff
@@ -707,6 +708,46 @@ def test_get_remote_data(mocker):
...
+    When
+        - run check_assigned_to_field command
+    Then
+        - Check that assign_to value is filled with the right email
+    """
+    class Client:
+        def get(self, table, value):
+            return USER_RESPONSE
+
+    assigned_to = {'link': 'https://dev113888.service-now.com/api/now/table/sys_user/oscar@example.com',
```
[the remark this answers is not in the document]
Comment (Shellyber) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:745]: This is the response from the ticket, getting link and value.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 3]
```diff
@@ -707,6 +708,46 @@ def test_get_remote_data(mocker):
...
+    """
+    Given:
+        -  Client class
+        -  Assigned_to field for user that doesn't exist in SNOW
+    When
+        - run check_assigned_to_field command
+    Then
+        - Check that assign_to value is empty
+    """
+    class Client:
```
[the remark this answers is not in the document]
Comment (Shellyber) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:721]: This is more simpler to use, we don't need the client as the rest of the test and all the mocks there. 
Just mock the get and the response we get.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 4]
```diff
@@ -2109,6 +2109,19 @@ def login_command(client: Client, args: Dict[str, Any]) -> Tuple[str, Dict[Any,
...
+    if assigned_to:
+        user_result = client.get('sys_user', assigned_to.get('value'))  # type: ignore[arg-type]
+        user = user_result.get('result', {})
+        if user:
+            user_email = user.get('email')
+            return user_email
+        else:
+            demisto.debug(f'Could not assign user {assigned_to.get("value")} since it does not exist in ServiceNow')
+            return ''
+    return
```
Comment (Itay4) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2122]: lets return either `None` or `''`, not both

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Itay4) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:721]: dont really see how it is simpler, will just lead to confusion for the next developer that will need to add tests here as we are not inconsistent

but if you insist, then ok

**Review by Itay4 — APPROVED** [review]

**Pull request merged**